### DISCUSSION TITLE
What would be the proper syntax to write this code?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#68041] [Help] [answered]
### Summary

From:

app>my>track>route.js

```
export async function GET (`/:id`, (req, res)) {

const pixelId = req.params.id;

res.send (`The pixel id is ${pixelId}`);
}
  
```

^^  the above code doesn't like the section "/:id", 

What am I trying to do? 

I have sent an email with the `src=http://localhost:3000/id/2`...   the idea is when the image tries to load, it will trigger the above route that will request the parameter from the src code and do some work from there.

So far I am getting the image in my Gmail, but nothing happens, likely because the syntax I have is bad.

**Comment by icyJoseph** [comment, marked as answer]
Refer to the docs: https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-route-segments

```ts
export async function GET(_, { params }) {
  const pixelId = params.id;
  return new Response(`The pixel id is ${pixelId}`);
}
```

And the path to the route.ts file, should be something like, `app/my/file/[id]/route.js`

And your fetch request, `fetch("/my/file/the-id-goes-here")` with a relative path from your own client. Or with the full path to the server from other systems.

Make sure you can reach this endpoint as desired with a proper tool first, like a browser, Postman, etc

**Reply by Elindo586** [reply]
Thanks.. it works..
